linker/shared/src/test/scala/org/scalajs/linker/testutils/TestIRBuilder.scala

```diff
@@ -50,10 +51,11 @@ object TestIRBuilder {
       jsNativeLoadSpec: Option[JSNativeLoadSpec] = None,
       memberDefs: List[MemberDef] = Nil,
       topLevelExportDefs: List[TopLevelExportDef] = Nil): ClassDef = {
-    ClassDef(ClassIdent(className), NON, kind, jsClassCaptures,
+    val notHashed = ClassDef(ClassIdent(className), NON, kind, jsClassCaptures,
         superClass.map(ClassIdent(_)), interfaces.map(ClassIdent(_)),
         jsSuperClass, jsNativeLoadSpec, memberDefs, topLevelExportDefs)(
         EOH)
+    Hashers.hashClassDef(notHashed)
```

Comment:
I've added a wip commit with a private implementation of SHA-1 to make this work. I've converted the PR to draft because I now need to write some tests for that `SHA1` object in isolation, to make sure it's consistent with the JVM.

I'll come back to this after Jan 4th, I guess, as I'll be without computer for 4 days, and then "busy with vacation" for 3 days.